dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/factory.py

```diff
@@ -512,52 +522,195 @@ def build_call_functions(self):
                 body = eq_mak.get_body()
-                body_tmp = []
-                for line in body:
-                    if "threadData" in line:
-                        line = line.replace("threadData,", "")
-                        body_tmp.append(line)
-                    #removing of clean ifdef
-                    elif "#ifdef" not in line and "#endif" not in line \
-                            and "SIM_PROF_" not in line and "NORETCALL" not in line:
-                        body_tmp.append(line)
+                body_tmp = self.handle_body_for_discrete(body, name_func_to_search, global_pattern_index)
+
                 if any(ext in " ".join(body_tmp) for ext in self.list_when_eq_to_filter):
                     continue
                 self.list_call_for_setz.extend(body_tmp)
 
+        for body in list_body_to_append_to_z:
+            body_tmp = self.handle_body_for_discrete(body, name_func_to_search, global_pattern_index)
+            self.list_call_for_setz.extend(body_tmp)
+
+
+    def handle_body_for_discrete(self, body, name_func_to_search, global_pattern_index):
+        list_omc_functions = self.reader.list_omc_functions
+        list_bool_var_names = [item.get_name() for item in self.list_all_bool_items]
+        body_tmp = []
+        for line in body:
+            ## When an integer is set by passing a reference to into a function parameter,
+            ## need to use an intermediate modelica_integer var to set the double value on Dynawo side
+            found = False
+            for name in name_func_to_search:
+                if name+"(" in line or name +" (" in line:
+                    for func in filter(lambda x: (x.get_name() == name), list_omc_functions):
+                        variables_to_replace = self.find_output_integer_double_vars(line, func)
+                        dic_var_name_to_temporary_name = {}
+                        (line, dic_var_name_to_temporary_name, global_pattern_index) = self.replace_var_names_by_temporary_and_build_dictionary(line, variables_to_replace, global_pattern_index)
+
+                        for name in dic_var_name_to_temporary_name.keys():
+                            test_param_address(name)
+                            if "integerDoubleVars" in to_param_address(name):
+                                body_tmp.append("  modelica_integer " + dic_var_name_to_temporary_name[name]+ " = (modelica_integer)" + to_param_address(name) + " /* " + name + "*/;\n")
+                            elif name in list_bool_var_names:
+                                body_tmp.append("  modelica_boolean " + dic_var_name_to_temporary_name[name]+ " = fromNativeBool(" + to_param_address(name) + " /* " + name + "*/);\n")
+                            else:
+                                body_tmp.append("  modelica_real " + dic_var_name_to_temporary_name[name]+ " = " + to_param_address(name) + " /* " + name + "*/;\n")
+
```

Comment:
Same kind of remark than for continuous variables - do we need to have first the declaration with assignment to the data->localData value and then the assignment by the omc function?